tests/src/core/testqgscompositionconverter.cpp

```diff
@@ -0,0 +1,427 @@
+     */
+    void importComposerTemplate();
+
+    /**
+     * Test import pictures from a composer template
+     */
+    void importComposerTemplatePicture();
+
+    /**
+     * Test import polygon from a composer template
+     */
+    void importComposerTemplatePolygon();
+
+    /**
+     * Test import polyline from a composer template
+     */
+    void importComposerTemplatePolyline();
+
+    /**
+     * Test import arrow from a composer template
+     */
+    void importComposerTemplateArrow();
+
+    /**
+     * Test import map from a composer template
+     */
+    void importComposerTemplateMap();
+
+    /**
+     * Test import legend from a composer template
+     */
+    void importComposerTemplateLegend();
+
+    /**
+     * Test import scalebar from a composer template
+     */
+    void importComposerTemplateScaleBar();
+
+
+  private:
+
+    void checkRenderedImage( QgsLayout *layout, const QString testName, const int pageNumber = 0 );
+
+    QDomElement loadComposition( const QString name );
+
+    QString mReport;
+
+};
+
+void TestQgsCompositionConverter::initTestCase()
+{
+  mReport = QStringLiteral( "<h1>Layout Tests</h1>\n" );
+  QgsSettings settings;
+  settings.setValue( QStringLiteral( "svg/searchPathsForSVG" ), QStringLiteral( TEST_DATA_DIR ) ) ;
+}
+
+void TestQgsCompositionConverter::cleanupTestCase()
+{
+  QString myReportFile = QDir::tempPath() + QDir::separator() + "qgistest.html";
+  QFile myFile( myReportFile );
+  if ( myFile.open( QIODevice::WriteOnly | QIODevice::Append ) )
+  {
+    QTextStream myQTextStream( &myFile );
+    myQTextStream << mReport;
+    myFile.close();
+  }
+}
+
+void TestQgsCompositionConverter::init()
+{
+
+}
+
+void TestQgsCompositionConverter::cleanup()
+{
+
+}
+
+
+void TestQgsCompositionConverter::importComposerTemplateLabel()
+{
+  QDomElement docElem( loadComposition( "2x_template_label.qpt" ) );
+  QgsReadWriteContext context;
+  QgsProject project;
+  QgsLayout *layout = QgsCompositionConverter::createLayoutFromCompositionXml( docElem, &project );
+
+  QVERIFY( layout );
+  QCOMPARE( layout->pageCollection()->pageCount(), 1 );
+
+  QList<QgsLayoutItemLabel *> items;
+  layout->layoutItems<QgsLayoutItemLabel>( items );
+  QVERIFY( items.size() > 0 );
+
+  //exportLayout( layout, QTest::currentTestFunction() );
+
+  // Check the label
+  const QgsLayoutItemLabel *label = items.at( 0 );
+  QVERIFY( label );
+  QCOMPARE( label->text(), QStringLiteral( "QGIS" ) );
+  QCOMPARE( label->pos().x(), 55.5333 );
+  QCOMPARE( label->pos().y(), 35.3929 );
+  QCOMPARE( label->sizeWithUnits().width(), 10.875 );
+  QCOMPARE( label->sizeWithUnits().height(), 6.0 );
+  QCOMPARE( label->referencePoint(), QgsLayoutItem::ReferencePoint::LowerRight );
+  QCOMPARE( label->frameStrokeColor(), QColor( 251, 0, 0, 255 ) );
+  QCOMPARE( label->frameStrokeWidth().length(), 0.2 );
+  QCOMPARE( ( int )label->rotation(), 4 );
+
+  checkRenderedImage( layout, QTest::currentTestFunction(), 0 );
```

Comment:
Can you please push up the font size on this label? The reference image is very small, and cross-platform rendering differences will make this test unstable.
Also need to make sure it uses the test reference font.

Reply:
The diff images are required to pass the test on Travis due to font differences, I've searched everywhere for documentation about the "safe" font for rendering tests on Travis but I couldn't find anything, additionally, existing tests apparently use a mix of "MS" and "Ubuntu" and "Courier" and "DejaVu" fonts so I could't guess what it was the "safe" one.

The test reference font to use is .... ?